test/widgets/message_list_test.dart

```diff
@@ -570,6 +571,23 @@ void main() {
             });
       });
 
+      testWidgets('markNarrowAsRead on mark-all-as-read when Unreads.oldUnreadsMissing: true', (tester) async {
+        const narrow = AllMessagesNarrow();
+        await setupMessageListPage(tester,
+          narrow: narrow, messages: [message], unreadMsgs: unreadMsgs);
+        check(isMarkAsReadButtonVisible(tester)).isTrue();
+        store.unreads.oldUnreadsMissing = true;
+
+        final connection = store.connection as FakeApiConnection;
```

Comment:
nit: there's an outer `connection` that this is equivalent to, in the same way as `store`:
```suggestion
```

(I see there's a bunch of existing cases of similar redundancy in this file. I'll send a quick cleanup for those.)

Reply:
bump